content/_includes/post.macros.njk

```diff
@@ -156,8 +156,17 @@ params:
       </header>
       {% if post.data.events %}
         {{ upcoming(post.data.events, links=false) }}
-      {% else %}
-        {{ post.data.summary | md | safe }}
+      {% elif post.data.summary %}
+        {% set maxWords = 50 %}
+        {% set summary = post.data.summary | md %}
+        {% set summaryPlain = summary | striptags %}
+        {% set summary = summaryPlain | elide(maxWords) | typogr
+          if (card != 'large') and (summaryPlain | wordcount > maxWords)
+          else summary
+        %}
```

Comment:
Yeah, I couldn't think of a good way to elide without stripping tags, or potentially breaking the html.

Reply:
Right. How many of our summaries contain markup?

Reply:
I don't know an easy way to count, but I'd guess ~1/3 or something? Maybe more.

Reply:
oh, actually - I think we need to strip links in summaries either way, since we use them in cards that are block-links… so maybe the `striptags` filter should be applied in both cases.

Reply:
Ok, I'm alright with that -- we may want to briefly review the relevant cases to be sure stripping markup doesn't change/remove meaning? But I'd rather be consistent, or find a way to elide without stripping markup.

Reply:
I did a quick search for blocks of code, and blockquotes - those seem like the main places we might lose something meaningful. Mostly I've just seen inline `code`, `em`/`strong`, and links. I think those are acceptable loss…

Leaning towards `striptags` on all "cards" but not on the large featured posts.

Reply:
Yeah, in reviewing I'm mostly seeing the same: links, strong/em, and code. I'm okay with this approach, but it does feel like a common problem... What about using a library e.g. https://github.com/oe/truncate-html?

Reply:
Yeah, that would make sense *except* we do have to strip links entirely in the linked cards. I think we likely also want to strip paragraphs…

The other option is going through manually to ensure all summaries use inline markdown without links?